BlueprintUI/Sources/Layout/ConstrainedPercent.swift

```diff
@@ -0,0 +1,85 @@
+import UIKit
+
+/// Constrains the size of the content element to a fraction of the parent's layout space.
+public struct ConstrainedSizeFraction: Element {
+    /// The element being constrained.
+    public var wrappedElement: Element
+    /// The fraction of the parent's horizontal layout space the content element should occupy. `nil` if the width is
+    /// unconstrained.
+    public var width: CGFloat?
+    /// The fraction of the parent's vertical layout space the content element should occupy. `nil` if the height is
+    /// unconstrained.
+    public var height: CGFloat?
+
+    /// Initializes with the given properties.
+    ///
+    /// - parameters:
+    ///   - width: The fraction of the parent's horizontal layout space the content element should occupy. `nil` if
+    ///   the width is unconstrained. By default, `nil`.
+    ///   - height: The fraction of the parent's vertical layout space the content element should occupy. `nil` if
+    ///   the height is unconstrained. By default, `nil`.
+    ///   - wrapping: The content element.
+    public init(width: CGFloat? = nil, height: CGFloat? = nil, wrapping wrappedElement: Element) {
+        self.width = width
```

Comment:
Do we want to validate the width and height is between 0 and 1? Could end up with some weirddd behavior if someone, say, passes in -1 or 2, etc. Precondition checks would be valuable.

Reply:
(either that, or clamp between 0 and 1)

Reply:
I'm open to this, and I considered it myself, but stopped short because I didn't know our general disposition here. I'll add a check.